src/lib/operators/join_hash.cpp

```diff
@@ -369,12 +390,19 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
           return;
         }
 
-        auto hashtable = std::make_shared<HashTable<LeftType>>(partition_size);
+        auto hashtable = std::make_shared<HashTable<HashType>>(partition_size);
 
         for (size_t partition_offset = partition_left_begin; partition_offset < partition_left_end;
              ++partition_offset) {
           auto& element = partition_left[partition_offset];
-          hashtable->put(element.value, element.row_id);
+
+          // use lexical cast if necessary
+          if constexpr (JoinHashTraits<LeftType, RightType>::needs_lexical_cast &&
+                        !std::is_same_v<LeftType, HashType>) {
+            hashtable->put(boost::lexical_cast<HashType>(element.value), element.row_id);
```

Comment:
Again that duplication. Can't we move that "check and cast if necessary" to its own method?

Reply:
done, but see my other comments above